spi/src/main/java/org/opensearch/ml/common/spi/tools/Tool.java

```diff
@@ -97,6 +97,18 @@ default boolean useOriginalInput() {
         return false;
     }
 
+    /**
+     * Get the tenant ID.
+     * @return the tenant ID
+     */
+    String getTenantId();
```

Comment:
Tenant ID will be a runtime variable, if a single instance of tool is shared between threads then this approach could lead to race conditions.